src/items.rs

```diff
@@ -790,6 +759,37 @@ impl<'a> FmtVisitor<'a> {
     }
 }
 
+fn rewrite_explicit_self(explicit_self: &ast::ExplicitSelf, args: &[ast::Arg]) -> String {
```

Comment:
I considered this, but reserved the `None` case for if we want to implement `Rewrite` for it later. I don't think `String::new` allocates, by the way.